src/app/workshops/workshops.component.ts

```diff
@@ -23,31 +24,84 @@ export class WorkshopsComponent {
 	private globalConstants:GlobalConstantsRepository;
 	
 	@ViewChild(WorkshopsListComponent) workshopsListChildComp:WorkshopsListComponent;
+	@ViewChild(WorkshopFilterComponent) workshopsFilterChildComp:WorkshopFilterComponent;
 
-	constructor(private globalConstantsRepository:GlobalConstantsRepository, private route:ActivatedRoute)
+	constructor(private globalConstantsRepository:GlobalConstantsRepository, private route:ActivatedRoute, private router:Router)
 	{
 		this.globalConstants = globalConstantsRepository;
-		this.hideFilter = true;
+        this.hideFilter = true;
+        
+        router.events.subscribe(event => {
+            if(event instanceof NavigationEnd)
+                {
+            this.setParameters();
+            this.updateUrl();
+                }
+        });
     }
 
     toggleFilterDropdown(event: any) {
         this.hideFilter = event;
-	}
-	
-	ngOnInit() {
-		var today = new Date();
-		this.startDate = `${today.getFullYear().toString()}/${(today.getMonth()+1).toString()}/${today.getDate().toString()}`;
-        this.endDate = `${(today.getFullYear()+10).toString()}/${(today.getMonth()+1).toString()}/${today.getDate().toString()}`;
-        let sub = this.route.params.subscribe(params => {
+    }
+    
+    setParameters() {
+        this.route.params.subscribe(params => {
             this.pageNumber = params['pageNumber'];
         });
-        
+
+        this.route.queryParams.subscribe(params => {
+            this.locationIdList = params['locations'];
+            this.categoryList = params['categories'];
+            this.minPrice = params['minPrice'];
+            this.maxPrice = params['maxPrice'];
+            this.startDate = params['startDate'];
+            this.endDate = params['endDate'];
+        });
+
+        var today = new Date();
+        if(!this.startDate)
+            {
+        this.startDate = `${today.getFullYear().toString()}/${(today.getMonth()+1).toString()}/${today.getDate().toString()}`;
+            }
+
+            if(!this.endDate)
+                {
+        this.endDate = `${(today.getFullYear()+10).toString()}/${(today.getMonth()+1).toString()}/${today.getDate().toString()}`;
+                }
+
+        this.workshopsFilterChildComp.setValuesFromParameters(this.minPrice, this.maxPrice, this.categoryList, this.locationIdList, this.startDate, this.endDate);
+    }
+	
+	ngOnInit() {
+        this.setParameters();
 		this.updateUrl();
-		}
+    }
+    
+    createUrl() : string {
+        let url = `/workshops/${this.pageNumber}?startDate=${this.startDate}&endDate=${this.endDate}`;
```

Comment:
helper functions for createUrl so that we can reuse in different components like workshop-list component

Reply:
was the plan. wanted to finish it faster.